### DISCUSSION TITLE
allow NVDA to interrupt speech when dynamic screen content refreshes

### DISCUSSION THREAD

**Discussion by coconut-catalina** [nvaccess/nvda#16403] [Ideas]
the idea is that when you are, for example working in the command line and the screen is constantly refreshing, such as using package managers, you don't want to have to interrupt speech by pressing control, this allows NVDA to interrupt what it was previously reading and immediately begin reading the refreshed content, not only does it come in handy in the command line, but on the web too because if NVDA is still reading previous screen content and a page has loaded, you have to wait until NVDA has finished reading the previous content before it begins reading the next one, so the setting could be something like, allow speech to be interrupted when screen context changes, it could be assigned to something like NVDA+enter, lets extend this further, if you only want screen context to be interrupted in terminal applications you could add the only in terminals toggle, so you would toggle the setting with NVDA+enter or another appropriate hotkey like this, allow speech to be interrupted when screen context changes on, on means than NVDA will interrupte speech in all contexts including web pages being loaded, in theory it should immediately read notifications immediately when you get them, the second option, only in terminals, EG, command prompt or powershell, or off, which is the classic behaviour

**Comment by lukaszgo1** [comment]
For consoles you can use https://github.com/mltony/nvda-console-toolkit/, an argument can certainly be made for implementing something similar into the core. I don't understand in which cases this can be useful on web pages - could you provide a step by step scenario and a sample web page?

**Comment by coconut-catalina** [comment]
Lets look at a tipicle  example, if you know someone who uses a Mac, and I do own a Mac, when a page loads and VoiceOver is active, VoiceOver immediately begins reading the new page, so ideally you want a feature where in that if a page has successfully loaded, NVDA stops reading, and starts reading the title of the page that just loaded, currently, NVDA keeps speaking while a web page loads and on browsers like chrome, there’s no indication that the page has loaded until NVDA sfinishes its previous read out, so you read the wikipedia article about Debian linux, you just got a notification but NVDA isn’t reading it write away because its busy reading the web page, having to go to the action centre is a pain, of course you could just wait it out but other screen readers immediately interrupt what they were doing and immediately start reading the new notification, again, VoiceOver on both mobile and desktop as well as prudence for android and super nova. Again this can be toggled off at will and is part of the main interrupt speech setting

**Reply by lukaszgo1** [reply]
These are two different scenarios:
- NVDA does not interrupt speech for notifications - this one is clear
- The web pages example is still unclear. When I'm on a page the only case in which page changes is when I move to a different one. Since to move you have to use a keyboard command which interrupts speech it is unclear what speech should be interrupted exactly. Again I encourage you to provide a sample web page and a list of steps describing what keys you press, what was spoken by NVDA and what you expected instead.

Feel free to create separate issues for notifications cancelling current speech and speech interruptions in terminal, as these two scenarios are defined well enough to be discussed via an issue.

**Comment by coconut-catalina** [comment]
Clearly I can’t provide a sample page, as that’s not part of the idea I had, this should work with any page, in summery for web pages, you want NVDA to immediately say the name of the newly loaded page, so lets say that NVDA is reading say, untitled—google chrome, even though a page has loaded or you create a new tab, you want that interrupted so that you can hear the name of the newly loaded page. Just look at how VoiceOver on the Mac works to get a clear understanding, or, a specific version of orca on linux, 45 to be precise

**Comment by josephsl** [comment]
INTEVENTION: let's try it again on a different discussion, complete with minimal to no spam, steps to reproduce, and a bit better rationale than comparing screen readers across different operating systems.

**Comment by coconut-catalina** [comment]
If this can’t be included in the core I understand but it would be a great feature

**Comment by josephsl** [comment]
Hi,

I’m sure there are ways to include this into NVDA Core. What we need are clearer use cases, including web sample, and for some, programs exhibiting this odd behavior. Also, let us try minimizing trolling and spamming here (as we’ve seen with some comments that are now gone).

Thanks.

**Comment by josephsl** [comment]
Provisionally reopening as long as trolls are dealt with.

**Comment by XLTechie** [comment]
For the webpage issue, it seems like you are asking for something to override 
expired speech.
E.g. if the title of a page changes, but NVDA is still saying or hasn't yet 
said, the old title, stop saying (or don't say) the old one, and just say the 
new one.

While that is a separate issue to the one first raised here, I could get behind 
that.

**Comment by Adriani90** [comment]
This already works with NVDA when using Microsoft Edge for example. But the page loading notification interupts the reporting of the window title.
I think #16381 will add a  quite good basis to fix such issues.
If the browser does not expose any notification when the page is loading, then NVDA will probably not be able to identify that the page is loading. I don't know how Voice Over can identify this.
